pkg/api/mongodb_repository.go

```diff
@@ -12,12 +16,56 @@ const (
 
 // MongoRepository represents a mongodb repository
 type MongoRepository struct {
-	session *mgo.Session
+	// TODO: we need to expose this so the plugins can use the same session. We should abstract the session and provide
+	// the plugins with a simple interface to search, insert, update and remove data from whatever backend implementation
+	Session     *mgo.Session
+	refreshTime time.Duration
 }
 
 // NewMongoAppRepository creates a mongo API definition repo
-func NewMongoAppRepository(session *mgo.Session) (*MongoRepository, error) {
-	return &MongoRepository{session}, nil
+func NewMongoAppRepository(dsn string, refreshTime time.Duration) (*MongoRepository, error) {
+	log.WithField("dsn", dsn).Debug("Trying to connect to MongoDB...")
+	session, err := mgo.Dial(dsn)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not connect to mongodb")
+	}
+
+	log.Debug("Connected to MongoDB")
+	session.SetMode(mgo.Monotonic, true)
+
+	return &MongoRepository{Session: session, refreshTime: refreshTime}, nil
+}
+
+// Close terminates the session.  It's a runtime error to use a session
+// after it has been closed.
+func (r *MongoRepository) Close() error {
+	r.Session.Close()
```

Comment:
shouldn't we `return r.Session.Close()`?

Reply:
`r.Session.Close()` from mongo returns void, so I can't return it